Sources/SPMBuildCore/PluginInvocation.swift

```diff
@@ -395,6 +401,8 @@ struct PluginScriptRunnerOutput: Codable {
         let environment: [String: String]?
         let inputPaths: [String]
         let outputPaths: [String]
-        let derivedSourcePaths: [String]
     }
+
+    var generatedOutputFiles: [String] = []
+    var prebuildOutputDirectories: [String] = []
```

Comment:
nit: should these be AbsolutePath?

Reply:
This is the struct that we deserialize from JSON, so this one currently has everything as strings so we'll be able to do our own validation with more custom messages when converting to the internal structure.  We could probably catch the deserialization error and figure out which field failed validation.  Be it seemed easier to keep everything as strings in the transport structure and then add more fine-grained checking in the internal structure that we actually use, where these are `AbsolutePath`.

Reply:
ah that sounds good as long as this is not used as the "model" and is here just for serialization purposes

Reply:
That's correct — the model uses `AbsolutePath`.